bandersnatch_vrfs/src/affine.rs

```diff
@@ -0,0 +1,102 @@
+use ark_ff::MontFp;
+use ark_ec::{short_weierstrass::{self, SWCurveConfig, SWFlags}, CurveConfig};
+use ark_serialize::{Compress, Read, SerializationError, Validate, Write};
+use ark_std::vec::Vec;
+use crate::bandersnatch::{BandersnatchConfig as BandersnatchConfigBase, SWAffine as AffineBase, SWProjective as ProjectiveBase};
+
+pub const COMPRESSED_POINT_SIZE: usize = 32;
+
+pub type BandersnatchAffine = short_weierstrass::Affine<BandersnatchConfig>;
+pub type BandersnatchProjective = short_weierstrass::Projective<BandersnatchConfig>;
+
+#[derive(Clone, Default, PartialEq, Eq)]
+pub struct BandersnatchConfig;
+
+const SW_GENERATOR_X: <BandersnatchConfig as CurveConfig>::BaseField =
+    MontFp!("30900340493481298850216505686589334086208278925799850409469406976849338430199");
+
+const SW_GENERATOR_Y: <BandersnatchConfig as CurveConfig>::BaseField =
+    MontFp!("12663882780877899054958035777720958383845500985908634476792678820121468453298");
+
+impl CurveConfig for BandersnatchConfig {
+    type BaseField = <BandersnatchConfigBase as CurveConfig>::BaseField;
+    type ScalarField = <BandersnatchConfigBase as CurveConfig>::ScalarField;
+
+    const COFACTOR: &'static [u64] = <BandersnatchConfigBase as CurveConfig>::COFACTOR;
+    const COFACTOR_INV: Self::ScalarField = <BandersnatchConfigBase as CurveConfig>::COFACTOR_INV;
+}
+
+impl SWCurveConfig for BandersnatchConfig {
+    const COEFF_A: Self::BaseField = <BandersnatchConfigBase as SWCurveConfig>::COEFF_A;
+    const COEFF_B: Self::BaseField = <BandersnatchConfigBase as SWCurveConfig>::COEFF_B;
+    const GENERATOR: BandersnatchAffine = BandersnatchAffine::new_unchecked(SW_GENERATOR_X, SW_GENERATOR_Y);
+    
+    #[inline(always)]
+    fn msm(bases: &[BandersnatchAffine], scalars: &[Self::ScalarField]) -> Result<BandersnatchProjective, usize> {
+        let bases: Vec<_> = bases.into_iter().map(|b| {
+            AffineBase { x: b.x, y: b.y, infinity: b.infinity }
+        }).collect();
+        BandersnatchConfigBase::msm(&bases, scalars).map(|p| {
+            BandersnatchProjective { x: p.x, y: p.y, z: p.z }
+        })
```

Comment:
Here and below a bit of unsafety would prevent copy of the slice in the Vec.
As the allowed backends are 2 (arkworks upstream and subtrate one) we can be quite confident that the memory layout is identical